multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/jobs/OrphanedDataCleaner.java

```diff
@@ -20,39 +21,43 @@
 public abstract class OrphanedDataCleaner<T extends AuditableConfiguration> implements Cleaner {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(OrphanedDataCleaner.class);
-    private static final String AUTH_ORIGIN = "uaa";
 
     private ApplicationConfiguration configuration;
     protected CloudControllerClient client;
     private boolean executed;
 
-    public OrphanedDataCleaner(ApplicationConfiguration applicationConfiguration) {
+    protected OrphanedDataCleaner(ApplicationConfiguration applicationConfiguration) {
         this.configuration = applicationConfiguration;
         this.executed = false;
     }
 
     @Override
     public void execute(Date expirationTime) {
         if (!executed) {
-            LOGGER.debug(CleanUpJob.LOG_MARKER, getStartCleanupLogMessage());
-            deleteOrphanedData();
-            LOGGER.debug(CleanUpJob.LOG_MARKER, getEndCleanupLogMessage());
+            LOGGER.info(CleanUpJob.LOG_MARKER, getStartCleanupLogMessage());
+            int deletedOrphanedDataCount = deleteOrphanedData();
+            LOGGER.info(CleanUpJob.LOG_MARKER, getEndCleanupLogMessage(deletedOrphanedDataCount));
             executed = true;
         }
     }
 
     protected abstract String getStartCleanupLogMessage();
 
-    protected abstract String getEndCleanupLogMessage();
+    protected abstract String getEndCleanupLogMessage(int deletedDataCount);
 
-    private void deleteOrphanedData() {
+    private int deleteOrphanedData() {
         List<T> configurationData = getConfigurationData();
-        configurationData.stream()
-                         .filter(this::hasNoAssociatedSpace)
-                         .peek(this::auditLogDeletion)
-                         .map(this::getSpaceId)
-                         .distinct()
-                         .forEach(this::deleteConfigurationDataBySpaceId);
+        List<String> spaceIDs = configurationData.stream()
+                                                 .filter(this::hasNoAssociatedSpace)
+                                                 .peek(this::auditLogDeletion)
+                                                 .map(this::getSpaceId)
+                                                 .distinct()
+                                                 .collect(Collectors.toList());
+        int deletedDataCount = 0;
+        for (String spaceId : spaceIDs) {
+            deletedDataCount += deleteConfigurationDataBySpaceId(spaceId);
+        }
+        return deletedDataCount;
```

Comment:
Utilize the stream API
configurationData.stream()
                         .filter(this::hasNoAssociatedSpace)
                         .peek(this::auditLogDeletion)
                         .map(this::getSpaceId)
                         .distinct()
                         .mapToInt(this::deleteConfigurationDataBySpaceId)
                         .sum();